test/contracts/kill_switch/kill_switch.test.js

```diff
@@ -0,0 +1,963 @@
+        defaultIssuesRegistry = IssuesRegistry.at(getNewProxyAddress(defaultRegistryReceipt))
+        await defaultIssuesRegistry.initialize()
+        await issuesRegistryACL.createPermission(securityPartner, defaultIssuesRegistry.address, SET_SEVERITY_ROLE, root, { from: root })
+      })
+
+      beforeEach('deploy DAO with a kill switch', async () => {
+        const receipt = await daoFactory.newDAOWithKillSwitch(root, defaultIssuesRegistry.address)
+        dao = Kernel.at(getEventArgument(receipt, 'DeployDAO', 'dao'))
+        acl = ACL.at(await dao.acl())
+        killSwitch = KillSwitch.at(await dao.killSwitch())
+
+        await acl.createPermission(root, dao.address, APP_MANAGER_ROLE, root, { from: root })
+        await acl.createPermission(owner, killSwitch.address, SET_DEFAULT_ISSUES_REGISTRY_ROLE, root, { from: root })
+        await acl.createPermission(owner, killSwitch.address, SET_ISSUES_REGISTRY_ROLE, root, { from: root })
+        await acl.createPermission(owner, killSwitch.address, SET_ALLOWED_INSTANCES_ROLE, root, { from: root })
+        await acl.createPermission(owner, killSwitch.address, SET_DENIED_BASE_IMPLS_ROLE, root, { from: root })
+        await acl.createPermission(owner, killSwitch.address, SET_HIGHEST_ALLOWED_SEVERITY_ROLE, root, { from: root })
+      })
+
+      beforeEach('create kill switched app', async () => {
+        const receipt = await dao.newAppInstance(SAMPLE_APP_ID, appBase.address, '0x', false, { from: root })
+        app = KillSwitchedApp.at(getNewProxyAddress(receipt))
+        await app.initialize(owner)
+      })
+
+      describe('integration', () => {
+        const itExecutesTheCall = () => {
+          it('executes the call', async () => {
+            await app.write(10, { from: owner })
+            assert.equal(await app.read(), 10)
+          })
+        }
+
+        context('when the function being called is not tagged', () => {
+          itExecutesTheCall()
+        })
+
+        context('when the function being called is tagged', () => {
+          const itAlwaysExecutesTheCall = () => {
+            context('when the instance being called is allowed', () => {
+              beforeEach('allow instance', async () => {
+                await killSwitch.setAllowedInstance(app.address, true, { from: owner })
+              })
+
+              context('when the base implementation is not denied', () => {
+                beforeEach('do not deny base implementation', async () => {
+                  await killSwitch.setDeniedBaseImplementation(appBase.address, false, { from: owner })
+                })
+
+                itExecutesTheCall()
+              })
+
+              context('when the base implementation is denied', () => {
+                beforeEach('deny base implementation', async () => {
+                  await killSwitch.setDeniedBaseImplementation(appBase.address, true, { from: owner })
+                })
+
+                itExecutesTheCall()
```

Comment:
Could we add a comment that allowing the instance has higher precedence than denying the base implementation?